syntax = "proto3";

package auth;

// Auth Service Definition
service AuthService {
    // Register a new user
    rpc Register(RegisterRequest) returns (AuthResponse);
    
    // Login and get JWT token
    rpc Login(LoginRequest) returns (AuthResponse);
    
    // Validate JWT token
    rpc ValidateToken(ValidateRequest) returns (ValidateResponse);
}

// Request Messages
message RegisterRequest {
    string username = 1;
    string password = 2;
    string role = 3;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message ValidateRequest {
    string token = 1;
}

// Response Messages
message AuthResponse {
    string status = 1;
    string message = 2;
    string token = 3;
    string user_id = 4;
    string role = 5;
}

message ValidateResponse {
    string status = 1;
    string message = 2;
    string user_id = 3;
    string role = 4;
    string username = 5;
}